frontend/src/pages/notebookController/screens/server/GPUSelectField.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { FormGroup, Select, SelectOption, Skeleton } from '@patternfly/react-core';
+import { getGPU } from '../../../../services/gpuService';
+import useNotification from '../../../../utilities/useNotification';
+
+type GPUSelectFieldProps = {
+  value: string;
+  setValue: (newValue: string) => void;
+};
+
+const GPUSelectField: React.FC<GPUSelectFieldProps> = ({ value, setValue }) => {
+  const [gpuDropdownOpen, setGpuDropdownOpen] = React.useState<boolean>(false);
+  const [gpuSize, setGpuSize] = React.useState<number>();
+  const [isFetching, setFetching] = React.useState(true);
+  const notification = useNotification();
+
+  React.useEffect(() => {
+    let lastCall = 0;
+    let cancelled = false;
+    const fetchGPU = () => {
+      setFetching(true);
+      lastCall = Date.now();
+      return getGPU().then((size) => {
+        if (cancelled) return;
+        setGpuSize(size || 0);
+        setFetching(false);
+      });
+    };
+
+    const errorCatch = (e: Error) => {
+      if (cancelled) return;
+      setFetching(false);
+      console.error(e);
+      notification.error('Failed to fetch GPU', e.message);
+    };
+
+    fetchGPU().catch(errorCatch);
+
+    const onUserClick = (): void => {
```

Comment:
oh, this is neat!  Perhaps in a future release we can put this higher up and add it to the global context so any component can check for "last activity"

Reply:
Yeah we could definitely have a global state of when the user was last doing something. This came out of a back and forth with Kyle. Seemed like a neat feature.